holmes/plugins/toolsets/grafana/toolset_grafana_tempo.py

```diff
@@ -553,6 +534,532 @@ def get_parameterized_one_liner(self, params: Dict) -> str:
         return f"{toolset_name_for_one_liner(self._toolset.name)}: Simple Tempo Traces Comparison"
 
 
+# New tools matching GrafanaTempoAPI methods
+
+
+class SearchTracesByQuery(Tool):
+    def __init__(self, toolset: BaseGrafanaTempoToolset):
+        super().__init__(
+            name="search_traces_by_query",
+            description=(
+                "Search for traces using TraceQL query language. "
+                "Uses the Tempo API endpoint: GET /api/search with 'q' parameter. "
+                'TraceQL allows complex filtering like: {resource.service.name="api"} && {span.http.status_code=500}'
+            ),
+            parameters={
+                "q": ToolParameter(
+                    description="TraceQL query (e.g., '{resource.service.name=\"api\" && span.http.status_code=500}')",
+                    type="string",
+                    required=True,
+                ),
+                "limit": ToolParameter(
+                    description="Maximum number of traces to return",
+                    type="integer",
+                    required=False,
+                ),
+                "start": ToolParameter(
```

Comment:
We only accept start/end time in timestamp format here, so this might not work when fetching traces from the time of a historical alert. (Fetch me all traces from the time of this alert.)

Could we add an eval for that please?